assets/javascripts/modules/checkout/validations.js

```diff
@@ -28,21 +38,49 @@ define(['$', 'modules/checkout/form-elements', 'modules/checkout/regex'], functi
     }
 
     var validatePersonalDetails = function () {
-        var emptyFields = mandatoryFieldsPersonalDetails.filter(function (field) {
-            var isEmpty = field.input.val() === '';
-            toggleError(field.container, isEmpty);
-            return isEmpty;
+
+        var staticValidations = new Promise(function (onSuccess, onFailure){
+            var emptyFields = mandatoryFieldsPersonalDetails.filter(function (field) {
+                var isEmpty = field.input.val() === '';
+                toggleError(field.container, isEmpty);
+                return isEmpty;
+            });
+            var noEmptyFields = emptyFields.length === 0;
+
+            var validEmail = regex.isValidEmail(form.$EMAIL.val());
+            renderEmailError(!validEmail);
+            toggleError(form.$EMAIL_CONTAINER, !validEmail);
+
+            var emailCorrectTwice = form.$EMAIL.val() === form.$CONFIRM_EMAIL.val();
+            toggleError(form.$CONFIRM_EMAIL_CONTAINER, validEmail && !emailCorrectTwice);
+
+            if(noEmptyFields && emailCorrectTwice && validEmail){
+                onSuccess();
+            }
+            else {
+                onFailure(new Error('Personal details incorrect.'));
+            }
         });
-        var noEmptyFields = emptyFields.length === 0;
 
-        var validEmail = regex.isValidEmail(form.$EMAIL.val());
-        renderEmailError(!validEmail);
-        toggleError(form.$EMAIL_CONTAINER, !validEmail);
+        var validationResult = staticValidations.then(function(){
```

Comment:
If you want to use a Promise pattern, rather than nesting promises you might want to try [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)